pennylane/devices/preprocess.py

```diff
@@ -162,7 +165,7 @@ def mid_circuit_measurements(
         mcm_method = "one-shot" if tape.shots else "deferred"
 
     if mcm_method == "one-shot":
-        return qml.dynamic_one_shot(tape)
+        return qml.dynamic_one_shot(tape, interface=interface)
```

Comment:
The `dynamic_one_shot` transform is created outside the ml boundary but applied in inner execute, after `convert_to_numpy_parameters`, which erases the original interface information. This allows the transform to remember what the interface was, and this is important for hardware like post selection.